go/test/endtoend/vtgate/schematracker/loadkeyspace/schema_load_keyspace_test.go

```diff
@@ -76,13 +77,27 @@ func TestLoadKeyspaceWithNoTablet(t *testing.T) {
 	for _, vttablet := range clusterInstance.Keyspaces[0].Shards[0].Vttablets {
 		err = vttablet.VttabletProcess.TearDown()
 		require.NoError(t, err)
+		timeout := time.After(1 * time.Minute)
+		down := false
+		for !down {
+			select {
+			case <-timeout:
+				t.Error("timeout - teardown of VTTablet")
+			case <-time.After(1 * time.Second):
+				down = vttablet.VttabletProcess.GetStatus() == ""
+			}
+		}
 	}
 
 	// Start vtgate with the schema_change_signal flag
 	clusterInstance.VtGateExtraArgs = []string{"--schema_change_signal"}
 	err = clusterInstance.StartVtgate()
 	require.NoError(t, err)
 
+	// After starting VTGate we need to leave enough time for resolveAndLoadKeyspace to reach
+	// the schema tracking timeout (5 seconds).
+	time.Sleep(10 * time.Second)
```

Comment:
Is there a way to use the same pattern you used above to wait on a status instead of a hardcoded sleep? We have seen scheduling delays in CI where 10 elapsed seconds may not be enough during heavy load ...

Reply:
I changed it to be a more dynamic timeout, which refreshes every second and timeout after 5 minutes. However, I couldn't come up with a way of checking the status. We know VTGate is up and running `StartVtgate` checks that for us.